src/main/java/com/movie_collection/gui/controllers/BaseController.java

```diff
@@ -0,0 +1,94 @@
+package com.movie_collection.gui.controllers;
+
+import com.google.inject.Inject;
+import com.movie_collection.bll.helpers.ViewType;
+import com.movie_collection.gui.controllers.abstractController.RootController;
+import com.movie_collection.gui.controllers.controllerFactory.IControllerFactory;
+import javafx.event.ActionEvent;
+import javafx.fxml.FXML;
+import javafx.scene.Parent;
+import javafx.scene.Scene;
+import javafx.scene.layout.StackPane;
+import javafx.stage.Modality;
+import javafx.stage.Stage;
+
+import java.io.IOException;
+
+/**
+ * Base controller navigation with switchable content
+ * Serves as a root for all the action :)
+ * injects the root controller
+ */
+public class BaseController extends RootController {
+
+    @FXML
+    private StackPane app_content;
+
+    @Inject
+    IControllerFactory rootController;
+
+    @Inject
+    public BaseController(IControllerFactory rootController) {
+        this.rootController = rootController;
+    }
+
+    /**
+     * method bound with fxml that loads the parent and shows new scene
+     * @param actionEvent event
+     */
+    @FXML
+    private void displayAllMovies(ActionEvent actionEvent) throws IOException {
```

Comment:
I have to rename this method it is incorrect naming with a purpose it should be onClickCreateMovie or smth cuz below I just specified view for creating and forgot to rename